src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -335,21 +365,26 @@ private void ImportLoadVar(int index, bool argument)
                 }
                 valueType = GetLLVMTypeForTypeDesc(type);
             }
-            else
+            else if (index < _locals.Length)
```

Comment:
I'd rather have an explicit parameter for whether something is a local or a temp since this is pretty subtle.